pkg/netbox/api/prefix_claim.go

```diff
@@ -53,16 +53,31 @@ func (r *NetboxClient) RestoreExistingPrefixByHash(hash string) (*models.Prefix,
 	}, nil
 }
 
-func isRequestingTheEntireParentPrefix(prefixClaim *models.PrefixClaim) (bool, error) {
+func isRequestingInvalidPrefixLength(prefixClaim *models.PrefixClaim, prefixFamily int64) (bool, string, error) {
 	parentPrefixSplit := strings.Split(prefixClaim.ParentPrefix, "/")
 	if len(parentPrefixSplit) != 2 {
-		return false, errors.New("invalid parent prefix format")
+		return false, "", errors.New("invalid parent prefix format")
 	}
 
-	if "/"+parentPrefixSplit[1] /* e.g. /24 */ == prefixClaim.PrefixLength /* e.g. /24 */ {
-		return true, nil
+	parentPrefixLength, err := strconv.Atoi(parentPrefixSplit[1])
+	if err != nil {
+		return false, "", err
+	}
+
+	requestedPrefixLength, err := strconv.Atoi(strings.TrimPrefix(prefixClaim.PrefixLength, "/"))
+	if err != nil {
+		return false, "", err
+	}
+
+	if parentPrefixLength == requestedPrefixLength {
+		return true, "requesting the entire parent prefix range is disallowed", nil
+	} else if parentPrefixLength > requestedPrefixLength {
+		return true, "requested prefix size must be smaller than the parent prefix size", nil
+	} else if prefixFamily == int64(IPv4Familiy) && requestedPrefixLength > 32 {
+		return true, "requested prefix length must be smaller than 32 for IPv4", nil
```

Comment:
Since we are on it, I think we can also add `requested prefix length must be smaller than 128 for IPv6`! :)

Reply:
Do we want to do the similar validation for IPv4

Reply:
If we can select the pattern with a condition we could do the validation in the type definition. As it is the same type and field for IPv4 and IPv6 I chose the validation that includes both of them.

Reply:
Makes sense! I checked the references it doesn't seem to allow conditional in the validation rules. 

Let's leave it as is :)